hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # can have more than one files
+        return True
+
+    class Meta:
+        verbose_name = 'Geographic Feature (ESRI Shapefiles)'
+        proxy = True
+
+processor_for(GeographicFeatureResource)(resource_processor)
+
+# define the GeographicFeatureMetaData metadata
+class GeographicFeatureMetaData(CoreMetaData):
+    geometryinformation = generic.GenericRelation(GeometryInformation)
+    fieldinformation = generic.GenericRelation(FieldInformation)
+    originalcoverage = generic.GenericRelation(OriginalCoverage)
+    originalfileinfo = generic.GenericRelation(OriginalFileInfo)
+
+    @classmethod
+    def get_supported_element_names(cls):
+        # get the names of all core metadata elements
+        elements = super(GeographicFeatureMetaData, cls).get_supported_element_names()
+        # add the name of any additional element to the list
+        elements.append('FieldInformation')
+        elements.append('OriginalCoverage')
+        elements.append('GeometryInformation')
+        elements.append('OriginalFileInfo')
+        return elements
+
+    def has_all_required_elements(self):
+        if not super(GeographicFeatureMetaData, self).has_all_required_elements():  # check required meta
+            return False
+        if not self.originalfileinfo.all().first():
+            return False
+        if not self.fieldinformation.all().first():
+            return False
+        if not self.originalcoverage.all().first():
+            return False
+        if not self.geometryinformation.all().first():
+            return False
+        if not (self.coverages.all().filter(type='box').first() or self.coverages.all().filter(type='point').first()):
+            return False
+        return True
+
+    def get_required_missing_elements(self):  # show missing required meta
+        missing_required_elements = super(GeographicFeatureMetaData, self).get_required_missing_elements()
+        if not self.coverages.all().filter(type='box').first():
+            missing_required_elements.append('Spatial Coverage: Box')
+        if not self.originalcoverage.all().first():
+            missing_required_elements.append('Spatial Reference')
+        if not self.geometryinformation.all().first():
+            missing_required_elements.append('Geometry Information')
+        if not self.originalfileinfo.all().first():
+            missing_required_elements.append('Original File Information')
+        # if not self.field_info.all().first():
+        #     missing_required_elements.append('FieldInformation')
+
+        return missing_required_elements
+
+    def get_xml(self):
+        from lxml import etree
+        # get the xml string representation of the core metadata elements
+        xml_string = super(GeographicFeatureMetaData, self).get_xml(pretty_print=False)
+
+        # create an etree xml object
+        RDF_ROOT = etree.fromstring(xml_string)
```

Comment:
I'm not going to ask for this here, but the fact that you have inheritance trees that keep parsing and re-serializing the same XML document up the class tree adding new elements to it is *really* inefficient and its something I'd like to see fixed down the road.